morphocluster/frontend/src/views/Files.vue

```diff
@@ -0,0 +1,196 @@
+                <div class="d-flex justify-content-center">
+                    <b-button size="sm" variant="primary" class="mx-2" @click.prevent="downloadFile">
+                        Download
+                    </b-button>
+                    <b-button size="sm" variant="primary" class="mx-2">
+                        Import as project
+                    </b-button>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import "@mdi/font/css/materialdesignicons.css";
+import * as api from "@/helpers/api.js";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "bootstrap";
+import { uploadFiles } from "../helpers/api.js";
+
+export default {
+    name: "FilesView",
+    props: { file_path: String },
+    components: {},
+    response: "",
+
+    data() {
+        return {
+            fields: [
+                { key: "name" },
+                "last_modified",
+            ],
+            files: [],
+            file: {},
+            fileInfos: [],
+            breadcrumb: null,
+            alerts: [],
+        };
+    },
+    created() {
+        this.initialize();
+    },
+    watch: {
+        $route: "initialize",
+    },
+    methods: {
+        getBasename(path) {
+            const parts = path.split('/');
+            return parts[parts.length - 1];
+        },
+        openFileInput() {
+            document.getElementById("fileInput").click();
+        },
+        async initialize() {
+            try {
+                const data = await api.getFileInfo(this.file_path);
+                const { children, breadcrumb_paths, ...fileinfo } = data;
+                this.breadcrumb = breadcrumb_paths;
+                this.files = children;
+                this.file = fileinfo;
+                this.fileInfos = [fileinfo];
+            } catch (error) {
+                console.error(error);
+                this.alerts.unshift({
+                    message: error.message,
+                    variant: "danger",
+                });
+            }
+        },
+        async handleDrop(event) {
+            this.uploadFiles(event, true);
+        },
+        async handleFileSelect(event) {
+            this.uploadFiles(event, false);
+        },
+        async uploadFiles(event, is_drop) {
+            event.preventDefault();
+            const selectedFiles = is_drop ? event.dataTransfer.files : event.target.files;
+            const formData = new FormData();
+            for (let i = 0; i < selectedFiles.length; i++) {
+                const file = selectedFiles[i];
+                formData.append('file', file);
+            }
+            const response = await uploadFiles(formData, this.file_path);
+            console.log("Data upload successful", response.message);
+            this.initialize();
+        },
+        downloadFile() {
+            if (this.fileInfos.length > 0) {
+                window.open(`/api/files/${this.file.path}` + "?download=1");
```

Comment:
We shouldn't need to build an url here. Use url_for in api.py to construct complete paths and leave no guesswork for the clients.